wordpress-mobile/WordPress-Android
# [Jetpack] Update playstore metadata

Pulls the Jetpack's playstore metadata (Store presence strings like `title`, `short description` and `long description`) from GlotPress for all the 16 translations we support.

ℹ️  Those were also just uploaded to the PlayStore, meaning that our store listing is now translated in those 16 locales. (\cc @leandroalonso for info btw)

💡  Note: this PR targets `release/17.8` because I'm hoping to make use of those during this cycle.

### How

* Updates the lanes in `Fastfile` to also pull the `title.txt`/`short_description.txt`/`full_description.txt` files from JP's GlotPress (for each GP locale) into `jetpack_metadata`
* Updates the list of locales used for Jetpack lanes
* Ensures source metadata (`en-US`) were copied too (as we can't use the same `en-gb` hack as for WordPress, see comments)

I then ran `fl download_jetpack_metadata_strings version:17.7 build_number:1072` to generate the various `.txt` files – that you can see in that PR – from the JP GlotPress translations.

> Note: This PR was created back when we were working on 17.7, which is why the metadata are from this 17.7 version even though this PR has since been rebased and retargeted onto `release/17.8`

<details><summary>PS: How I used Fastlane <tt>supply</tt> to automate the metadata update on PlayStore</summary>

I used fastlane's `supply` command line to upload the title, short description and long description into the PlayStore for all new 16 locales automatically for me.

```bash
$ bundle exec fastlane supply --package_name com.jetpack.android --metadata_path fastlane/jetpack_metadata/android --json_key ./google-upload-credentials.json --skip_upload_changelogs true --skip_upload_apk true --skip_upload_aab true
```

Note: This was the occasion to do a proof of concept that paaHJt-2ga-p2 should be pretty easy to implement into the `Fastfile`, though I decided to keep that for a separate PR as this will require some cleanup to do it nicely (I prefer to keep things separate for easier-to-review PRs).

</details>

### To Test

* Cut a temporary branch from this one, and push it to the remote
* Run `fastlane download_metadata_strings`
* Check it still downloads the changelogs, title, and descriptions from GlotPress for both WP and JP, and pushes two commits (one for WP, one for JP). Check the txt metadata files generated.
* Delete the temp branch

### Notes

* I only ran the `download_jetpack_metadata_strings` lane (ie for Jetpack) and didn't commit the changes made by the one for WordPress, to avoid polluting the PR with WordPress 17.7's changelog `.txt` files being updated – as the goal of this PR is to focus on Jetpack.
* ~The short description for `de-DE` (German) ended up being too long (88 chars) – see pxLjZ-6w8-p2. This is why `de-DE/short_description.txt` doesn't appear in that PR.~ Solved since.
